syncano/models/fields.py

```diff
@@ -504,11 +505,15 @@ def validate(self, value, model_instance):
         super(ModelField, self).validate(value, model_instance)
 
         if not isinstance(value, (self.rel, dict)):
-            raise self.ValidationError('Value needs to be a {0} instance.'.format(self.rel.__name__))
+            if not self.is_data_object_mixin:
```

Comment:
Maybe `if not isinstance(value, (self.rel, dict)) and not self.is_data_object_mixin:`?